src/HostnameValidator.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace ipl\Validator;
+
+use ipl\I18n\Translation;
+
+/**
+ * Validates Host name and it's IP address
+ */
+class HostnameValidator extends BaseValidator
+{
+    use Translation;
+
+    /**
+     * Returns true if and only if $value is a valid email address
+     * according to RFC2822
+     *
+     * @link   http://www.ietf.org/rfc/rfc2822.txt RFC2822
+     * @link   http://www.columbia.edu/kermit/ascii.html US-ASCII characters
+     * @param string $value
+     * @return boolean
+     */
+    public function isValid($value)
+    {
+        $this->clearMessages();
+
+        //decode IDN domain name if possible
+        if (function_exists('idn_to_ascii')) {
+            $value = idn_to_ascii($value);
+        }
+
+        return filter_var($value, FILTER_VALIDATE_DOMAIN, FILTER_FLAG_HOSTNAME);
```

Comment:
This does not always return a boolean. Also, this validator doesn't set any message in case of failure.